src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -247,7 +248,26 @@ public void Insert(int index, T item)
             /// Adds an item to the <see cref="ICollection{T}"/>.
             /// </summary>
             /// <param name="item">The object to add to the <see cref="ICollection{T}"/>.</param>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public void Add(T item)
+            {
+                int count = _count;
+                T[] elements = _elements;
+
+                if ((uint)count < (uint)elements.Length)
+                {
+                    elements[count] = item;
+                    _count = count + 1;
+                }
+                else
+                {
+                    AddWithResize(item);
+                }
+            }
+
+            // Improve code quality as uncommon path
```

Comment:
I don't understand what this code comment means. Is this a TODO item that we need to improve the code quality? What does "as uncommon path" have to do with it?

Reply:
The `AddWithResize`-path is called from `Add` and is on the cold-path / uncommon-path. The comment is to "explain" why no-inlining is added. Similar to https://github.com/dotnet/coreclr/pull/9539#discussion_r100715659

Reply:
How about:

```cs
// Specify NoInlining so that we are guaranteed an opportunity to service this method
```